### ISSUE TITLE
No selection feedback for shell menu items

### ISSUE THREAD

**Issue by Magendanz** [xamarin/Xamarin.Forms#5816]
### Description
Unlike ShellItems, MenuItems don't get selection feedback or indicate selected state after being tapped.  The flyout is also not automatically hidden after selection when FlyoutBehavior if set to Flyout.

### Steps to Reproduce

1.  Create new project based on Xaramin shell template
2.  Modify default shell page to enable items in flyout menu

`<Shell xmlns="http://xamarin.com/schemas/2014/forms" 
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" 
       xmlns:local="clr-namespace:XamShellApp.Views"
       RouteScheme="app"
       Route="xamshellapp"
       FlyoutBehavior="Flyout"
       Title="XamShellApp"
       x:Class="XamShellApp.AppShell"
       x:Name="MyShell">

    
    <Shell.Items>
        <ShellItem Title="Items 1">
            <ShellSection>
                <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
            </ShellSection>
        </ShellItem>
        <ShellItem Title="Items 2">
            <ShellSection>
                <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
            </ShellSection>
        </ShellItem>
    </Shell.Items>
    <Shell.MenuItems>
        <MenuItem BindingContext="{x:Reference MyShell}" Command="{Binding AboutCommand}" Text="About" />
    </Shell.MenuItems>
</Shell>`

Note: Binding context on MenuItem must be explicitly set until bug #5707 is fixed.

3.  Add Command to handle menu selection

`using Xamarin.Forms;
using XamShellApp.Views;

namespace XamShellApp
{
    public partial class AppShell : Shell
    {
        public Command AboutCommand { get; }

        public AppShell()
        {
            Routing.RegisterRoute("about", typeof(AboutPage));
            AboutCommand = new Command(() =>
            {
                GoToAsync("app:///xamshellapp/about");
                FlyoutIsPresented = false;
            });

            BindingContext = this;
            InitializeComponent();
        }
    }
}`

### Expected Behavior

Selecting a menu items should behave the same as selecting a ShellItem.

### Actual Behavior

Selecting a menu item:
- Has no visual feedback when tapped
- Does not maintain selected state like ShellItem
- Does not automatically hide flyout after selection

### Basic Information

- Version with issue: v4.0.0.204370-pre8
- Last known good version: N/A
- IDE: v16.0.0
- Platform Target Frameworks: 
  - Android: v9.2.0.5 SDK 
  - iOS:  Not tested (but expect same issue)

**Comment by samhouts** [comment]
@Magendanz Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by Magendanz** [comment]
Here you go...
[Xamarin.Forms Bug #5816.zip](https://github.com/xamarin/Xamarin.Forms/files/3049228/Xamarin.Forms.Bug.5816.zip)

**Comment by samhouts** [comment]
You're quick!! Thanks! We'll take a look.

**Comment by samhouts** [comment]
@Magendanz Did you add `FlyoutIsPresented = false;` to workaround the Flyout not closing automatically? Or does the Flyout still not close for you even with that there? It closes for me with that code, so I want to make sure we're seeing the same thing.

**Comment by Magendanz** [comment]
Yes, that was my workaround.  It seems to me that selecting a menu item should automatically hide the flyout unless FlyoutBehavior is Locked.  This is what happens in regular menus (e.g. menu bars, context menus, hierarchical menus, menu buttons) when you select a menu item.  You never have to explicitly close those.

**Comment by davidortinau** [comment]
Related to #4509

**Comment by samhouts** [comment]
still an issue in pre10

**Comment by Magendanz** [comment]
Still an issue in v4.5.0.617...over a year later!

**Comment by demoh2019** [comment]
+1
